stateful-functions-flink/stateful-functions-flink-core/src/main/java/com/ververica/statefun/flink/core/StatefulFunctionsUniverseValidator.java

```diff
@@ -32,5 +32,8 @@ void validate(StatefulFunctionsUniverse statefulFunctionsUniverse) {
     if (statefulFunctionsUniverse.functions().isEmpty()) {
       throw new IllegalStateException("There are no function providers defined.");
     }
+    if (statefulFunctionsUniverse.sinks().isEmpty()) {
```

Comment:
In a `Stateful Functions` application, sinks (egresses) are optional.

Reply:
> In a `Stateful Functions` application, sinks (egresses) are optional.

But in Translator, if sinks is empty, this will cause NPE.

Reply:
This is correct, that is why the second change you propose (in `EgressToSinkTranslator`) is accepted. But consider the case where a stateful function application does not define any egresses (which is legal) then your change (in `StatefulFunctionsUniverseValidator`) will make it illegal.

Reply:
@igalshilman You are right. I misunderstood egress validation. By the way, what does `StatefulFunctionsUniverseValidator` comment that `// TODO: complete this` mean?

Reply:
And I have already remove `StatefulFunctionsUniverseValidator` sink validation.